subprojects/packagefiles/check/src/meson.build

```diff
@@ -1,12 +1,11 @@
-if get_option('default_library') == 'shared' and cc.get_id() == 'msvc'
-  ck_dll_exp_interface = '-DCK_DLL_EXP=__declspec(dllimport)'
-  ck_dll_exp_private = '-DCK_DLL_EXP=__declspec(dllexport)'
+if host_machine.system() == 'windows'
+  ck_dll_exp_interface = '-DCK_DLL_EXP=@0@'.format(get_option('default_library') != 'static' ? '__declspec(dllimport)' : '""')
+  ck_dll_exp_private = '-DCK_DLL_EXP=@0@'.format(get_option('default_library') != 'static' ? '__declspec(dllexport)' : '""')
 else
-  ck_dll_exp_interface = '-DCK_DLL_EXP='
-  ck_dll_exp_private = '-DCK_DLL_EXP='
+  ck_dll_exp_interface = '-DCK_DLL_EXP=__attribute__((visibility("default")))'
```

Comment:
The pkg-config / declare_dependency value actually doesn't need or want to be defined, simply set it `= []` as a no-op to avoid uselessly confusing command lines.

Reply:
Not sure I understand the confusing command line comment.

Reply:
A quick edit of such a thing looks even more confusing:

```
--- a/subprojects/packagefiles/check/src/meson.build
+++ b/subprojects/packagefiles/check/src/meson.build
@@ -1,8 +1,8 @@
 if host_machine.system() == 'windows'
-  ck_dll_exp_interface = '-DCK_DLL_EXP=@0@'.format(get_option('default_library') != 'static' ? '__declspec(dllimport)' : '""')
-  ck_dll_exp_private = '-DCK_DLL_EXP=@0@'.format(get_option('default_library') != 'static' ? '__declspec(dllexport)' : '""')
+  ck_dll_exp_interface = get_option('default_library') != 'static' ? '__declspec(dllimport)' : '""')
+  ck_dll_exp_private = get_option('default_library') != 'static' ? '__declspec(dllexport)' : '""')
 else
-  ck_dll_exp_interface = '-DCK_DLL_EXP=__attribute__((visibility("default")))'
+  ck_dll_exp_interface = ''
   ck_dll_exp_private = '-DCK_DLL_EXP=__attribute__((visibility("default")))'
 endif
 
@@ -35,7 +35,7 @@ checklib = library(
   'check',
   sources,
   gnu_symbol_visibility: 'hidden',
-  c_args: ['-DHAVE_CONFIG_H=1', ck_dll_exp_private],
+  c_args: ['-DHAVE_CONFIG_H=1', '-DCK_DLL_EXP=@0@'.format(ck_dll_exp_private)],
   include_directories: [confinc],
   dependencies: [libm, compatlib_dep],
 )
@@ -43,5 +43,5 @@ checklib = library(
 check_dep = declare_dependency(
   include_directories: '.',
   link_with: checklib,
-  compile_args: ck_dll_exp_interface,
+  compile_args: '-DCK_DLL_EXP=@0@'.format(ck_dll_exp_interface),
 )

```

Reply:
But that's also not what I suggested so I don't know why that bothers you.

Reply:
I think this is what you meant now?

Reply:
Yes, exactly. :)